### PR TITLE
[SIG-1902] Supporting components

### PR DESCRIPTION
**Pull request by janjaap**
This PR adds the following components:
- ### BackLink
  _Component with customisable URL and content_

  Screenshot 2019-11-12 at 08 06 09
  
  Screenshot 2019-11-12 at 08 49 11

- ### FormAlert
  _Notification block that shows a title and, optionally, a message_

  FormAlert, just title

  FormAlert, title and message

  FormAlert, notification style, just title

  FormAlert, notification style, just title

- ### FormFooter
  _Absolute positioned container that can show a cancel button, a reset button and/or a submit button_

FormAlert, notification style, title and message

- ### Input
  _Implementation of the `@datapunt/asc-ui` `<Input />` component with the addition of a hint text_

  Screenshot 2019-11-12 at 09 07 39

### In addition
It also updates the packages `@testing-library/jest-dom` and `@testing-library/react` to their latest stable versions.

### CODE REVIEW

File: src/components/BackLink/index.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+  </StyledLink>
+);
+
+BackLink.defaultProps = {
+  className: '',
+};
+
+BackLink.propTypes = {
+  /** The BackLink label */
+  children: PropTypes.node.isRequired,
```
Comment (jasperswart) [inline src/components/BackLink/index.js]: that is incorrect comment isn't it?

Comment (janjaap) [reply src/components/BackLink/index.js]: The `children` prop can be considered the component's label, but I guess the term `label` is a bit ambiguous. I'll change it.

File: src/components/BackLink/__tests__/Backlink.test.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    const { container, getByText } = render(
+      withAppContext(<BackLink to={to}>{label}</BackLink>)
+    );
+
+    expect(getByText(label)).toBeInTheDocument();
+
+    const link = container.querySelector('a');
+    expect(link).toBeInTheDocument();
+    expect(link.getAttribute('href')).toEqual(to);
+  });
```
Comment (jasperswart) [inline src/components/BackLink/__tests__/Backlink.test.js]: there is a test missing for clicking the link

File: src/components/FormAlert/index.js [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+      margin-bottom: 0;
+    `}
+`;
+
+/**
+ * Component that will display a title and, optionally, a message block. Should be used
+ * for generic feedback that is not related to individual fields.
+ */
+const FormAlert = ({ className, message, title, isNotification }) => (
+  <Div className={className} isNotification={isNotification}>
```
Comment (jasperswart) [inline src/components/FormAlert/index.js]: this should be called Alert I think there is also FormError component in DS.

it should have 4 variants:
[https://designsystem.amsterdam.nl/7awj1hc9f/p/901ede ](url)

I don't see those

Comment (janjaap) [reply src/components/FormAlert/index.js]: What should be called `Alert`? The Component that is the default export or the `<Div>` component on line 45?

This particular component is not meant as a generic notification component, but is specific to the context of a form. That is why this `FormAlert` component doesn't have the four different states that are in the design system. It is not necessary for the current scope. If needed, we can change the component at a later date, but for now, I'm not going to bother investing time in functionality that we don't yet need.

File: src/components/BackLink/__tests__/Backlink.test.js (hunk 2 above)
Comment (janjaap) [reply src/components/BackLink/__tests__/Backlink.test.js]: What should the test verify in your opinion? The element is an anchor; there is no need that the browser does what it is supposed to be doing.

File: src/components/Input/index.js [hunk 4]
```diff
@@ -0,0 +1,62 @@
+import React, { Fragment } from 'react';
+import PropTypes from 'prop-types';
```
Comment (jasperswart) [inline src/components/Input/index.js]: I would wait with this implementation of this component until we have a new form lib.
we should create a wrapper component for all types of form components with a label and a subtitle and the field.
but if it is needed now just add it.

File: src/components/Input/index.js [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+import styled, { css } from 'styled-components';
+import {
+  Input as Field,
+  Label as FieldLabel,
+  styles,
+  themeColor,
+  Typography,
+  themeSpacing,
+} from '@datapunt/asc-ui';
```
Comment (jasperswart) [inline src/components/Input/index.js]: I miss a input placeholder here

Comment (janjaap) [reply src/components/Input/index.js]: Not sure what you mean. There is no `Placeholder` component in `@datapunt/asc-ui`. If you mean setting a placeholder on the input element, that is already possible by passing that prop, like so
```jsx
  <Input placeholder="Foo bar baz" />
```

Line 44 takes care of that.

File: src/components/Input/index.js (hunk 4 above)
Comment (janjaap) [reply src/components/Input/index.js]: We're going to need the `asc-ui` implementation anyway, regardless of which lib we choose.

File: src/components/Input/index.js (hunk 5 above)
Comment (jasperswart) [reply src/components/Input/index.js]: yes true but I the test is still <Input placeholder="Foo bar baz" /> missing

File: src/components/BackLink/__tests__/Backlink.test.js (hunk 2 above)
Comment (jasperswart) [reply src/components/BackLink/__tests__/Backlink.test.js]: the link should link to the right place that is part of the component

Comment (jasperswart) [reply src/components/BackLink/__tests__/Backlink.test.js]: we should have a talk about the with Carlos about what to test and what not in general

File: src/components/FormAlert/index.js (hunk 3 above)
Comment (janjaap) [reply src/components/FormAlert/index.js]: IMHO, `Notification` is too generic a name for a component that is meant to show feedback related to actions performed in forms.

**Pull request merged**